doc/specs/stdlib_hashmaps.md

```diff
@@ -0,0 +1,2088 @@
+      do i=1, 15
+          dummy_val % value(i) = i
+      end do
+      allocate(other_in, source=dummy_val)
+      call copy_other( other_in, other_out )
+      call free_other( other_out )
+    end program demo_free_other
+```
+
+
+#### `get` - extracts the data from a derived type
+
+##### Status
+
+Experimental
+
+##### Description
+
+Extracts the data from a `key_type` or `other_type` and stores it
+in the variable `value`.
+
+##### Syntax
+
+`call [[stdlib_hashmap_wrappers:get]]( key, value )`
+
+or
+
+`call [[stdlib_hashmap_wrappers:get]]( other, value )`
+
+##### Class
+
+Subroutine.
+
+##### Argument
+
+`key`: shall be a scalar expression of type `key_type`. It
+is an `intent(in)` argument.
+
+`other`: shall be a scalar expression of type `other_type`. It
+is an `intent(in)` argument.
+
+`value`: if the the first argument is of `key_type` `value` shall be
+an allocatable default character string variable, or 
+an allocatable vector variable of type integer and kind `int8`,
+otherwise the first argument is of `other_type` and `value` shall be
+an allocatable of `class(*)`. It is an `intent(out)` argument.
+
+##### Example
+
+```fortran
+    program demo_get
+      use stdlib_hashmap_wrappers, only: &
+          get, key_type, set
+      use iso_fortran_env, only: int8
+      implicit none
+      integer(int8), allocatable :: value(:), result(:)
+      type(key_type) :: key
+      integer(int_8) :: i
+      allocate( value(1:15) )
+      do i=1, 15
+        value(i) = i
+      end do
+      call set( key, value )
+      call get( key, result )
+      print *, `RESULT == VALUE = ', all( value == result )
+    end program demo_get
+```
+
+
+#### `hasher_fun`- serves aa a function prototype.
+
+##### Status
+
+Experimental
+
+##### Description
+
+Serves as a prototype for hashing functions with a single, `key`,
+argument of type `key_type` returning an `int32` hash value.
+
+##### Syntax
+
+`type([[stdlib_hashmap_wrappers:hasher_fun]]), pointer :: fun_pointer`
+
+##### Class
+
+Pure function prototype
+
+##### Argument
+
+`key`: Shall be a rank one array expression of type `integer(int8)`.
+It is an `intent(in)` argument.
+
+##### Result character
+
+The result is a scalar integer of kind `int32`.
+
+##### Result value
+
+The result is a hash code.
+
+##### Note
+
+`hasher_fun` is a prototype for defining dummy arguments and function
+pointers intended for use as a hash function for the hash maps.
+
+##### Example
+
+```fortran
+    program demo_hasher_fun
+      use stdlib_hashmap_wrappers, only: &
+          fnv_1a_hasher, hasher_fun, set
+      use iso_fortran_env, only: int8, int32 
+      implicit none
+      type(hasher_fun), pointer :: hasher_pointer
+      integer(int8), allocatable :: array1(:)
+      integer(int32) :: hash
+      type(key_type) :: key
+      hasher_pointer => fnv_1a_hasher
+      array1 = [ 5_int8, 4_int8, 3_int8, 1_int8, 10_int8, 4_int8 ]
+      call set( key, array1 )
+      hash = hassher_pointer(key)
```

Comment:
```suggestion
      hash = hasher_pointer(key)
```

Reply:
I have corrected it in my version and will push it later.